### PR TITLE
Add comparison feature to the evaluation tool. Fix lcs.

### PR DESCRIPTION
**Pull request by mgsergio**
@ygorshenin

### CODE REVIEW

File: tools/python/openlr/quality.py [hunk 1]
```diff
@@ -74,21 +74,41 @@ def lcs(l1, l2, eq=operator.eq):
     diff.extend(reversed(l2[:j]))
     return common[::-1], diff[::-1]
 
+def almost_equal(s1, s2, Δ=1e-5):
```
Comment (ygorshenin) [inline tools/python/openlr/quality.py]: Please, don't. I guess this is the time to stop using unicode.

File: tools/python/openlr/quality.py [hunk 2]
```diff
@@ -167,5 +199,40 @@ def calculate(tree):
...
+            'Base' if μ1 - μ2 > 0 else 'New',
+            μ1 - μ2
+        ))
+    else:
+        print('{}\t{}'.format(
+            'segment_id', 'intersection_weight')
+        )
+        for x in assessed_scores.items():
+            print('{}\t{}'.format(*x))
+        print('μ: {:.4f}, σ: {:.4f}'.format(
```
Comment (ygorshenin) [inline tools/python/openlr/quality.py]: mean and variance, please.

File: tools/python/openlr/quality.py [hunk 3]
```diff
@@ -131,21 +151,33 @@ def parse_segments(tree, limit):
...
+    for s in dst.findall('Segment'):
+        assert not s.find('GoldenRoute')
+        try:
+            golden_route = golden_routes[int(s.find('.//ReportSegmentID').text)]
+            if golden_route:
+                s.append(golden_route)
+        except KeyError:
+            continue
+
+def mean(xs):
```
Comment (ygorshenin) [inline tools/python/openlr/quality.py]: We have numpy for that. Please, use it.

File: tools/python/openlr/quality.py (hunk 1 above)
Comment (mgsergio) [reply tools/python/openlr/quality.py]: Done.

File: tools/python/openlr/quality.py (hunk 2 above)
Comment (mgsergio) [reply tools/python/openlr/quality.py]: Done.

File: tools/python/openlr/quality.py (hunk 3 above)
Comment (mgsergio) [reply tools/python/openlr/quality.py]: Done.

**Comment by mgsergio** [comment]
PTAL

File: tools/python/openlr/quality.py [hunk 4]
```diff
@@ -3,6 +3,7 @@
 
 import xml.etree.ElementTree as ET
 import operator
+import numpy as np
```
Comment (ygorshenin) [inline tools/python/openlr/quality.py]: Sort?

File: tools/python/openlr/quality.py [hunk 5]
```diff
@@ -74,21 +75,41 @@ def lcs(l1, l2, eq=operator.eq):
...
+    >>> a = (LatLon(55.89259, 37.72521), LatLon(55.89269, 37.72535))
+    >>> b = (LatLon(55.89259, 37.72522), LatLon(55.8927, 37.72536))
+    >>> almost_equal(a, b)
+    True
+    >>> a = (LatLon(55.89259, 37.72519), LatLon(55.89269, 37.72535))
+    >>> b = (LatLon(55.89259, 37.72522), LatLon(55.8927, 37.72536))
+    >>> almost_equal(a, b)
+    False
+    """
+    eps = 2 * eps
```
Comment (ygorshenin) [inline tools/python/openlr/quality.py]:
```python
eps *= 2
```
?

File: tools/python/openlr/quality.py [hunk 6]
```diff
@@ -167,5 +191,40 @@ def calculate(tree):
...
+            print('{}\t{}\t{}\t{}'.format(
+                seg_id,
+                assessed_scores[seg_id], candidate_scores[seg_id],
+                assessed_scores[seg_id] - candidate_scores[seg_id]
+            ))
+        mean1 = np.mean(list(assessed_scores.values()))
+        variance1 = np.std(list(assessed_scores.values()), ddof=1)
+        mean2 = np.mean(list(candidate_scores.values()))
+        variance2 = np.std(list(candidate_scores.values()), ddof=1)
+        print('Base: mean: {:.4f}, variance: {:.4f}'.format(mean1, variance1))
```
Comment (ygorshenin) [inline tools/python/openlr/quality.py]: s/variance/std/

File: tools/python/openlr/quality.py [hunk 7]
```diff
@@ -167,5 +191,40 @@ def calculate(tree):
...
+            'segment_id', 'A', 'B', 'Diff')
+        )
+        for seg_id in assessed_scores:
+            print('{}\t{}\t{}\t{}'.format(
+                seg_id,
+                assessed_scores[seg_id], candidate_scores[seg_id],
+                assessed_scores[seg_id] - candidate_scores[seg_id]
+            ))
+        mean1 = np.mean(list(assessed_scores.values()))
+        variance1 = np.std(list(assessed_scores.values()), ddof=1)
```
Comment (ygorshenin) [inline tools/python/openlr/quality.py]: s/variance1/std1/g
s/variance2/std2/g

File: tools/python/openlr/quality.py (hunk 4 above)
Comment (mgsergio) [reply tools/python/openlr/quality.py]: Done.

File: tools/python/openlr/quality.py (hunk 5 above)
Comment (mgsergio) [reply tools/python/openlr/quality.py]: Done.

File: tools/python/openlr/quality.py (hunk 6 above)
Comment (mgsergio) [reply tools/python/openlr/quality.py]: Done.

File: tools/python/openlr/quality.py (hunk 7 above)
Comment (mgsergio) [reply tools/python/openlr/quality.py]: Done.

**Comment by mgsergio** [comment]
PTAL

**Pull request merged**